packages/ember-routing-htmlbars/lib/keywords/closure-action.js

```diff
@@ -45,6 +45,12 @@ export default function closureAction(morph, env, scope, params, hash, template,
         if (!action) {
           throw new EmberError(`An action named '${actionName}' was not found in ${target}.`);
         }
+      } else if (actionType === 'function') {
+        // on-change={{action setName}}
```

Comment:
`setName` here is a function as apposed to `"setName"` which is a string and an action name. I can change to `setNameFunc` if needed. In another comment, I will describe our use case. Maybe, it's not worth allowing targets for functions.